### PR TITLE
Linq samples for restriction

### PR DESCRIPTION
**Pull request by matthewreily**
Added a couple of more examples and refactored the sample methods to be static instead of instance.
Issue #103 cc @BethMassi

### CODE REVIEW

**Comment by BethMassi** [comment]
LGTM. @cartermp and I discussed the static vs. instance methods last week and we were leaning on the instance method approach, but static does make the program.cs easier on the eyes. We should be consistent I think though across the samples. If I get another thumbs up I'll merge and change the samples readme. /cc @blackdwarf

**Comment by matthewreily** [comment]
Great.  I'll be happy to update all the samples either way you want to go.

File: samples/linq/csharp/restriction/Where-Sample-4.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+            string[] digits = { "zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine" };
+
+            var shortDigits = digits.Where((digit, index) => digit.Length < index);
+
+            Console.WriteLine("Short digits:");
+            foreach (var d in shortDigits)
+            {
+                Console.WriteLine("The word {0} is shorter than its value.", d);
+            }
+		}
```
Comment (cartermp) [inline samples/linq/csharp/restriction/Where-Sample-4.cs]: Tabbing here is a bit off.

**Comment by cartermp** [comment]
LGTM modulo the one piece of feedback.  Thanks a lot @matthewreily!

**Comment by matthewreily** [comment]
Pushed fix for tabbing issues @cartermp Thanks!

**Comment by cartermp** [comment]
Thanks @matthewreily, much appreciated.  Mind rebasing to squash that last commit?  I'll merge once that's taken care of.

**Comment by matthewreily** [comment]
@cartermp I attempted to squash the commit... Could you verify it was done correctly please?

**Comment by cartermp** [comment]
Looks good.  Thanks again!

**Pull request merged**